test/database.ts

```diff
@@ -2216,4 +2220,166 @@ describe('Database', () => {
+        return [operations, {}];
+      };
+
+      const [results] = await database.getOperations();
+      assert.deepStrictEqual(results, operations);
+    });
+
+    it('should create filter for querying the database in combination with user supplied filter', async () => {
+      const operations: IOperation[] = [{name: 'my-operation'}];
+
+      database.instance.getDatabaseOperations = async options => {
+        assert.strictEqual(
+          options.filter,
+          `(name:${DATABASE_FORMATTED_NAME}) AND (someOtherAttribute: aValue)`
+        );
+        return [operations, {}];
+      };
+
+      const [results] = await database.getOperations({
+        filter: 'someOtherAttribute: aValue',
+      });
+      assert.deepStrictEqual(results, operations);
+    });
+
+    it('should options with given gaxOptions', async () => {
+      const operations: IOperation[] = [{name: 'my-operation'}];
+      const gaxOpts = {
+        timeout: 1000,
+      };
+
+      database.instance.getDatabaseOperations = async options => {
+        assert.strictEqual(options.gaxOptions, gaxOpts);
+        return [operations, {}];
+      };
+
+      const [results] = await database.getOperations({
+        filter: 'someOtherAttribute: aValue',
+        gaxOptions: gaxOpts,
+      });
+      assert.deepStrictEqual(results, operations);
+    });
+  });
+
+  describe('restore', () => {
+    const BACKUP_NAME = 'backup-name';
+    const BACKUP_FORMATTED_NAME =
+      INSTANCE.formattedName_ + '/backups/' + BACKUP_NAME;
+
+    it('should make the correct request', async () => {
+      const QUERY = {};
+      const ORIGINAL_QUERY = extend({}, QUERY);
+      const expectedReqOpts = extend({}, QUERY, {
+        databaseId: NAME,
+        parent: INSTANCE.formattedName_,
+        backup: BACKUP_FORMATTED_NAME,
+      });
+
+      database.id = NAME;
+      database.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'restoreDatabase');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+
+        assert.notStrictEqual(config.reqOpts, QUERY);
+        assert.deepStrictEqual(QUERY, ORIGINAL_QUERY);
+      };
+
+      await database.restore(BACKUP_FORMATTED_NAME);
+    });
+
+    it('should accept a backup name', async () => {
+      const QUERY = {};
+      const expectedReqOpts = extend({}, QUERY, {
+        databaseId: NAME,
+        parent: INSTANCE.formattedName_,
+        backup: BACKUP_FORMATTED_NAME,
+      });
+
+      database.id = NAME;
+      database.request = config => {
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+      };
+
+      await database.restore(BACKUP_NAME);
+    });
+
+    it('should accept gaxOpts and a callback', async () => {
+      const options = {
+        timeout: 1000,
+      };
+
+      database.request = config => {
+        assert.deepStrictEqual(config.gaxOpts, options);
```

Comment:
Let's use the `done` callback in this test so we can place it after this line. That way, we know the assertion was reached.